functests/sctp/sctp.go

```diff
@@ -45,59 +49,136 @@ var _ = Describe("sctp", func() {
+						return runningPod.Status.Phase
+					}, 1*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodFailed))
+				})
 
-			By("Connecting a client to the server")
-			clientArgs := []string{"-ip", runningPod.Status.PodIP, "-port", "30101", "-lport", "30102"}
-			clientPod := scptTestPod("sctpclient", clientNode, "sctpclient", clientArgs)
-			client.Client.Pods(testNamespace).Create(clientPod)
+				AfterEach(func() {
+					namespaces.Clean("sctptest", client.Client)
+					deleteService(client.Client)
+				})
+			})
+		})
+		var _ = Describe("Test Connectivity", func() {
+			var activeService *k8sv1.Service
+			var runningPod *k8sv1.Pod
+			var serverPod *k8sv1.Pod
+			var nodes *k8sv1.NodeList
+			var err error
 
-			Eventually(func() k8sv1.PodPhase {
-				pod, err := client.Client.Pods(testNamespace).Get(serverPod.Name, metav1.GetOptions{})
+			execute.BeforeAll(func() {
+				checkForSctpReady(client.Client)
+			})
+			BeforeEach(func() {
+				By("Choosing the nodes for the server and the client")
+				nodes, err = client.Client.Nodes().List(metav1.ListOptions{
+					LabelSelector: sctpNodeSelector,
+				})
+				Expect(err).ToNot(HaveOccurred())
+				clientNode = nodes.Items[0].ObjectMeta.Labels[hostnameLabel]
+				serverNode = nodes.Items[0].ObjectMeta.Labels[hostnameLabel]
+				if len(nodes.Items) > 1 {
+					serverNode = nodes.Items[1].ObjectMeta.Labels[hostnameLabel]
+				}
+				activeService = createSctpService(client.Client)
+				By("Starting the server")
+				serverArgs := []string{"-ip", "0.0.0.0", "-port", "30101", "-server"}
+				pod := scptTestPod("scptserver", serverNode, "sctpserver", serverArgs)
+				pod.Spec.Containers[0].Ports = []k8sv1.ContainerPort{
+					k8sv1.ContainerPort{
+						Name:          "sctpport",
+						Protocol:      k8sv1.ProtocolSCTP,
+						ContainerPort: 30101,
+					},
+				}
+				serverPod, err = client.Client.Pods(testNamespace).Create(pod)
 				Expect(err).ToNot(HaveOccurred())
-				return pod.Status.Phase
-			}, 1*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodSucceeded))
+				By("Fetching the server's ip address")
+				Eventually(func() k8sv1.PodPhase {
+					runningPod, err = client.Client.Pods(testNamespace).Get(serverPod.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return runningPod.Status.Phase
+				}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+
+			})
+			Context("Client Server Connection", func() {
+				// OCP-26759
+				It("Kernel Module is loaded", func() {
+					checkForSctpReady(client.Client)
+				})
+				// OCP-26760
+				It("Should connect a client pod to a server pod", func() {
+					testClientServerConnection(client.Client, runningPod.Status.PodIP,
+						activeService.Spec.Ports[0].Port, clientNode, serverPod.Name)
+				})
+				// OCP-26763
+				It("Should connect a client pod to a server pod. Feature LatencySensitive Active", func() {
+					fg, err := client.Client.FeatureGates().Get("cluster", metav1.GetOptions{})
```

Comment:
@nikita133 the problem here is that the server exits as soon as it receives a message (see https://github.com/openshift-kni/cnf-features-deploy/blob/master/tools/sctptester/main.go#L83 )
Since we are using the same server pod, the second test will not find it running and the client will fail.